streamer/src/nonblocking/quic.rs

```diff
@@ -905,21 +910,36 @@ impl ConnectionTable {
         }
     }
 
-    fn remove_connection(&mut self, key: ConnectionTableKey, port: u16) -> bool {
+    fn remove_connection(
+        &mut self,
+        key: ConnectionTableKey,
+        port: u16,
+        maybe_stable_id: Option<usize>,
+    ) -> usize {
         if let Entry::Occupied(mut e) = self.table.entry(key) {
             let e_ref = e.get_mut();
             let old_size = e_ref.len();
-            e_ref.retain(|connection| connection.port != port);
+
+            e_ref.retain(|connection_entry| {
+                connection_entry.port != port
+                    || maybe_stable_id
```

Comment:
This seems a little tough to read. Maybe a comment explaining conditions where entry is removed (port matches, and stable ID is either none or matches)